From ff9cb0bdf4c7cf631534726b7608616c200d9ede Mon Sep 17 00:00:00 2001
From: Zhi Zhou <zhi.zhou@amlogic.com>
Date: Fri, 18 Sep 2015 05:58:16 -0400
Subject: [PATCH 5851/5965] PD#112840 changed some other dtd for codec memory
 manger

based the commit:

PD#112840: amports:codec_mm:merged codec mermory manager from 3.14

Change-Id: I0bbaaf34a40d41db9fb8a762697a74723870af56

Change-Id: I70988549ec26539dc69288bf43168db977cf3054
---
 arch/arm/boot/dts/amlogic/meson8_k100_1G.dtd  | 23 +++++++--
 arch/arm/boot/dts/amlogic/meson8_k100_2G.dtd  | 32 +++++++++---
 arch/arm/boot/dts/amlogic/meson8_k200a_1G.dtd | 24 +++++++--
 .../dts/amlogic/meson8_k200a_1G_hdmiin.dtd    | 25 ++++++++--
 arch/arm/boot/dts/amlogic/meson8_k200a_2G.dtd | 24 +++++++--
 .../boot/dts/amlogic/meson8_k200a_2G_4k2k.dtd | 23 +++++++--
 .../dts/amlogic/meson8_k200a_2G_hdmiin.dtd    | 24 +++++++--
 .../dts/amlogic/meson8_k200b_1G_emmc_sdhc.dtd | 25 ++++++++--
 .../meson8_k200b_1G_emmc_sdhc_hdmiin.dtd      | 27 ++++++++--
 .../dts/amlogic/meson8_k200b_1G_emmc_sdio.dtd | 25 ++++++++--
 .../meson8_k200b_1G_emmc_sdio_hdmiin.dtd      | 27 ++++++++--
 .../boot/dts/amlogic/meson8_k200b_2G_4k2k.dtd | 34 +++++++++++--
 .../dts/amlogic/meson8_k200b_2G_emmc_sdhc.dtd | 34 +++++++++++--
 .../dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd | 34 +++++++++++--
 .../dts/amlogic/meson8_k200b_2G_hdmiin.dtd    | 33 +++++++++++--
 arch/arm/boot/dts/amlogic/meson8b_m100_1G.dtd | 22 +++++++--
 .../boot/dts/amlogic/meson8b_m101_512M.dtd    | 22 +++++++--
 arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd | 22 +++++++--
 arch/arm/boot/dts/amlogic/meson8b_m200_1G.dtd | 22 +++++++--
 .../boot/dts/amlogic/meson8b_m200_1G_iptv.dtd | 30 ++++++++++--
 arch/arm/boot/dts/amlogic/meson8b_m201_1G.dtd | 28 +++++++++--
 arch/arm/boot/dts/amlogic/meson8b_skt.dtd     |  3 +-
 .../boot/dts/amlogic/meson8m2_n200C_1G.dtd    | 23 +++++++--
 .../boot/dts/amlogic/meson8m2_n200C_2G.dtd    | 24 +++++++--
 .../arm/boot/dts/amlogic/meson8m2_n200_1G.dtd |  4 +-
 .../arm/boot/dts/amlogic/meson8m2_n200_2G.dtd | 31 ++++++++++--
 arch/arm/boot/dts/amlogic/meson8m2_pxp.dtd    | 32 ++++++++----
 arch/arm/boot/dts/amlogic/meson8m2_skt.dtd    | 22 +++++++--
 arch/arm/boot/dts/amlogic/mesong9bb_n302.dtd  | 41 ++++++++++------
 arch/arm/boot/dts/amlogic/mesong9bb_n303.dtd  | 40 +++++++++------
 arch/arm/boot/dts/amlogic/mesong9bb_n304.dtd  | 26 ++++++++--
 .../boot/dts/amlogic/mesong9bb_t826_pxp.dtd   | 49 ++++++++++++-------
 .../boot/dts/amlogic/mesong9bb_t826_skt.dtd   | 49 ++++++++++++-------
 .../boot/dts/amlogic/mesong9bb_t828_skt.dtd   | 49 ++++++++++++-------
 arch/arm/boot/dts/amlogic/mesong9tv_n210.dtd  | 46 +++++++++++------
 arch/arm/boot/dts/amlogic/mesong9tv_n300.dtd  | 44 +++++++++++------
 .../boot/dts/amlogic/mesong9tv_n300_15g.dtd   | 45 +++++++++++------
 .../boot/dts/amlogic/mesong9tv_n300_1g.dtd    | 49 ++++++++++++-------
 arch/arm/boot/dts/amlogic/mesong9tv_n301.dtd  | 44 +++++++++++------
 arch/arm/boot/dts/amlogic/mesong9tv_pxp.dtd   | 48 +++++++++++-------
 arch/arm/boot/dts/amlogic/mesong9tv_skt.dtd   | 49 ++++++++++++-------
 41 files changed, 952 insertions(+), 326 deletions(-)

diff --git a/arch/arm/boot/dts/amlogic/meson8_k100_1G.dtd b/arch/arm/boot/dts/amlogic/meson8_k100_1G.dtd
index d0d8308c577d..e14152de591f 100644
--- a/arch/arm/boot/dts/amlogic/meson8_k100_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k100_1G.dtd
@@ -156,7 +156,9 @@ void root_func(){
 		reserve-iomap = "true";
 
 	};
-
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -167,8 +169,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x00a00000>;//10M,you can use 8M if memory isn't enough
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -181,10 +183,21 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-                reserve-memory = <0x04000000>;//64M
-		reserve-iomap = "true";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
 
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		///linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
 /// ***************************************************************************************
 ///	-	DISP&MM-PostProcess
 //$$ MODULE="DISP&MM-PostProcess"
diff --git a/arch/arm/boot/dts/amlogic/meson8_k100_2G.dtd b/arch/arm/boot/dts/amlogic/meson8_k100_2G.dtd
index 8e42639c01ab..f26d3431fa8c 100644
--- a/arch/arm/boot/dts/amlogic/meson8_k100_2G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k100_2G.dtd
@@ -85,6 +85,14 @@ void root_func(){
         #address-cells = <1>;
         #size-cells = <1>;
 
+//$$ DEVICE="mem_4k2k_decoder"
+//$$ L3 PROP_U32 2 ="reg"
+		mem_4k2k_decoder: region@1 {
+			region_name = "cma_4k2k";
+			linux,contiguous-region;
+			reg = <0x00000000 0x0ca00000>;
+		};
+		
         cma_0:region@0 {
            region_name = "cma_0";
            reg = <0 0x01800000>;
@@ -155,6 +163,9 @@ void root_func(){
 
 	};
 
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -165,12 +176,10 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x2000000>;
-              reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
-
-
 /// ***************************************************************************************
 ///     -       DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V video dec"
@@ -181,10 +190,21 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-                reserve-memory = <0x10000000>;
-                reserve-iomap = "true";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
 
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		//linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
 /// ***************************************************************************************
 ///	-	DISP&MM-PostProcess
 //$$ MODULE="DISP&MM-PostProcess"
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200a_1G.dtd b/arch/arm/boot/dts/amlogic/meson8_k200a_1G.dtd
index 3f3289d95f77..35e26c681f57 100644
--- a/arch/arm/boot/dts/amlogic/meson8_k200a_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200a_1G.dtd
@@ -158,6 +158,9 @@ void root_func(){
 		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -168,7 +171,7 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
+		//reserve-memory = <0x02000000>; // 32M
 		//reserve-iomap = "true";
 	};
 
@@ -182,10 +185,22 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-								reserve-memory = <0x04000000>; // 64M
-//		reserve-iomap = "true";
-		linux,contiguous-region = <&mem_4k2k_decoder>;
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
+    
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
 //$$ DEVICE="picdec"
@@ -211,6 +226,7 @@ void root_func(){
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 /// ***************************************************************************************
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200a_1G_hdmiin.dtd b/arch/arm/boot/dts/amlogic/meson8_k200a_1G_hdmiin.dtd
index c7d0d13d4d1b..cf0e940874d3 100644
--- a/arch/arm/boot/dts/amlogic/meson8_k200a_1G_hdmiin.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200a_1G_hdmiin.dtd
@@ -158,6 +158,9 @@ void root_func(){
 		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -168,8 +171,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -182,10 +185,21 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-								reserve-memory = <0x04000000>; // 64M
-//		reserve-iomap = "true";
-		linux,contiguous-region = <&mem_4k2k_decoder>;
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
 //$$ DEVICE="picdec"
@@ -211,6 +225,7 @@ void root_func(){
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 /// ***************************************************************************************
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200a_2G.dtd b/arch/arm/boot/dts/amlogic/meson8_k200a_2G.dtd
index 38d6e8b91b0a..bc0c49b07d9b 100644
--- a/arch/arm/boot/dts/amlogic/meson8_k200a_2G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200a_2G.dtd
@@ -151,6 +151,9 @@ void root_func(){
 		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -161,8 +164,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -175,9 +178,21 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-								reserve-memory = <0x10000000>; // 256M
-								reserve-iomap = "true";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
 //$$ DEVICE="picdec"
@@ -203,6 +218,7 @@ void root_func(){
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 /// ***************************************************************************************
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200a_2G_4k2k.dtd b/arch/arm/boot/dts/amlogic/meson8_k200a_2G_4k2k.dtd
index ee43db4d8925..1b13560d16b9 100644
--- a/arch/arm/boot/dts/amlogic/meson8_k200a_2G_4k2k.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200a_2G_4k2k.dtd
@@ -151,6 +151,9 @@ void root_func(){
 		reserve-memory = <0x01800000>; //8x1920x1088x3/2=24M
 	};
 
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -161,8 +164,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -175,10 +178,21 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-                reserve-memory = <0x10000000>;
-                reserve-iomap = "true";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
 
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
 /// ***************************************************************************************
 ///	-	DISP&MM-PostProcess
 //$$ MODULE="DISP&MM-PostProcess"
@@ -190,6 +204,7 @@ void root_func(){
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 /// ***************************************************************************************
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200a_2G_hdmiin.dtd b/arch/arm/boot/dts/amlogic/meson8_k200a_2G_hdmiin.dtd
index c44d625afbec..00936d04a4bb 100644
--- a/arch/arm/boot/dts/amlogic/meson8_k200a_2G_hdmiin.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200a_2G_hdmiin.dtd
@@ -151,6 +151,9 @@ void root_func(){
 		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -161,8 +164,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -175,9 +178,21 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-								reserve-memory = <0x10000000>; // 256M
-								reserve-iomap = "true";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
 //$$ DEVICE="picdec"
@@ -203,6 +218,7 @@ void root_func(){
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 /// ***************************************************************************************
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc.dtd
index 6e1ec67512f6..f071f1ab8d9a 100644
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc.dtd
@@ -160,6 +160,10 @@ void root_func(){
 		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -170,7 +174,7 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
+		//reserve-memory = <0x02000000>; // 32M
 		//reserve-iomap = "true";
 	};
 
@@ -184,10 +188,22 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-								reserve-memory = <0x04000000>; // 64M
-//		reserve-iomap = "true";
-		linux,contiguous-region = <&mem_4k2k_decoder>;
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
+
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
 //$$ DEVICE="picdec"
@@ -213,6 +229,7 @@ void root_func(){
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 /// ***************************************************************************************
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc_hdmiin.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc_hdmiin.dtd
index beee0588fffd..8a286f5333b7 100644
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc_hdmiin.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc_hdmiin.dtd
@@ -160,6 +160,10 @@ void root_func(){
 		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -170,8 +174,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -184,10 +188,22 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-								reserve-memory = <0x04000000>; // 64M
-//		reserve-iomap = "true";
-		linux,contiguous-region = <&mem_4k2k_decoder>;
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
+
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
 //$$ DEVICE="picdec"
@@ -213,6 +229,7 @@ void root_func(){
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 /// ***************************************************************************************
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio.dtd
index 1dc8488f2d68..f2c252cf1970 100644
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio.dtd
@@ -160,6 +160,10 @@ void root_func(){
 		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -170,7 +174,7 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
+		//reserve-memory = <0x02000000>; // 32M
 		//reserve-iomap = "true";
 	};
 
@@ -184,10 +188,22 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-								reserve-memory = <0x04000000>; // 64M
-//		reserve-iomap = "true";
-		linux,contiguous-region = <&mem_4k2k_decoder>;
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
+
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
 //$$ DEVICE="picdec"
@@ -213,6 +229,7 @@ void root_func(){
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 /// ***************************************************************************************
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio_hdmiin.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio_hdmiin.dtd
index 0708b563ab89..8326fda44a19 100644
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio_hdmiin.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio_hdmiin.dtd
@@ -160,6 +160,10 @@ void root_func(){
 		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -170,8 +174,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -184,10 +188,22 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-								reserve-memory = <0x04000000>; // 64M
-//		reserve-iomap = "true";
-		linux,contiguous-region = <&mem_4k2k_decoder>;
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
+
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
 //$$ DEVICE="picdec"
@@ -213,6 +229,7 @@ void root_func(){
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 /// ***************************************************************************************
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_4k2k.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_4k2k.dtd
index 0c669f93474b..478d153de7a4 100644
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_4k2k.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_4k2k.dtd
@@ -84,7 +84,13 @@ void root_func(){
 
         #address-cells = <1>;
         #size-cells = <1>;
-
+//$$ DEVICE="mem_4k2k_decoder"
+//$$ L3 PROP_U32 2 ="reg"
+		mem_4k2k_decoder: region@1 {
+			region_name = "cma_4k2k";
+			linux,contiguous-region;
+			reg = <0x00000000 0x0ca00000>;
+		};
         cma_0:region@0 {
            region_name = "cma_0";
            reg = <0 0x01200000>;
@@ -150,6 +156,10 @@ void root_func(){
 		reserve-memory = <0x01800000>; //8x1920x1088x3/2=24M
 	};
 
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -160,8 +170,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -174,10 +184,23 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-                reserve-memory = <0x10000000>;
-                reserve-iomap = "true";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
 
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
+
+
 /// ***************************************************************************************
 ///	-	DISP&MM-PostProcess
 //$$ MODULE="DISP&MM-PostProcess"
@@ -189,6 +212,7 @@ void root_func(){
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 /// ***************************************************************************************
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdhc.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdhc.dtd
index a4931e87db17..3e3f570332b5 100644
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdhc.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdhc.dtd
@@ -85,7 +85,13 @@ void root_func(){
 
         #address-cells = <1>;
         #size-cells = <1>;
-
+//$$ DEVICE="mem_4k2k_decoder"
+//$$ L3 PROP_U32 2 ="reg"
+		mem_4k2k_decoder: region@1 {
+			region_name = "cma_4k2k";
+			linux,contiguous-region;
+			reg = <0x00000000 0x0ca00000>;
+		};
         cma_0:region@0 {
            region_name = "cma_0";
            reg = <0 0x01200000>;
@@ -151,6 +157,10 @@ void root_func(){
 		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -161,8 +171,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -175,9 +185,22 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-                reserve-memory = <0x10000000>;
-                reserve-iomap = "true";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
+
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
 //$$ DEVICE="picdec"
@@ -203,6 +226,7 @@ void root_func(){
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 /// ***************************************************************************************
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd
index 5556937e2017..9f8a29e6002e 100644
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd
@@ -85,7 +85,13 @@ void root_func(){
 
         #address-cells = <1>;
         #size-cells = <1>;
-
+//$$ DEVICE="mem_4k2k_decoder"
+//$$ L3 PROP_U32 2 ="reg"
+		mem_4k2k_decoder: region@1 {
+			region_name = "cma_4k2k";
+			linux,contiguous-region;
+			reg = <0x00000000 0x0ca00000>;
+		};
         cma_0:region@0 {
            region_name = "cma_0";
            reg = <0 0x01200000>;
@@ -151,6 +157,10 @@ void root_func(){
 		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -161,8 +171,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -175,9 +185,22 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-                reserve-memory = <0x10000000>;
-                reserve-iomap = "true";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
+
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
 //$$ DEVICE="picdec"
@@ -203,6 +226,7 @@ void root_func(){
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 /// ***************************************************************************************
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_hdmiin.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_hdmiin.dtd
index 2af656bc3261..6cf2eadae92c 100644
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_hdmiin.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_hdmiin.dtd
@@ -84,7 +84,13 @@ void root_func(){
 
         #address-cells = <1>;
         #size-cells = <1>;
-
+//$$ DEVICE="mem_4k2k_decoder"
+//$$ L3 PROP_U32 2 ="reg"
+		mem_4k2k_decoder: region@1 {
+			region_name = "cma_4k2k";
+			linux,contiguous-region;
+			reg = <0x00000000 0x0ca00000>;
+		};
         cma_0:region@0 {
            region_name = "cma_0";
            reg = <0 0x01200000>;
@@ -150,6 +156,10 @@ void root_func(){
 		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -160,8 +170,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -174,10 +184,22 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-                reserve-memory = <0x10000000>;
-                reserve-iomap = "true";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
 
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
+
 /// ***************************************************************************************
 ///	-	DISP&MM-PostProcess
 //$$ MODULE="DISP&MM-PostProcess"
@@ -189,6 +211,7 @@ void root_func(){
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 /// ***************************************************************************************
diff --git a/arch/arm/boot/dts/amlogic/meson8b_m100_1G.dtd b/arch/arm/boot/dts/amlogic/meson8b_m100_1G.dtd
index 6b0b7e2d9d02..e91e4e815708 100644
--- a/arch/arm/boot/dts/amlogic/meson8b_m100_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_m100_1G.dtd
@@ -155,6 +155,9 @@ void root_func(){
         display_size_default = <768 1024 768 3072 32>;//768x1024x4x3==0x0900000 < 0x01000000
 	};
 
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -165,8 +168,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x00a00000>;//10M,you can use 8M if memory isn't enough
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -179,10 +182,21 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-                reserve-memory = <0x04000000>;
-		reserve-iomap = "true";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
 
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+	
 /// ***************************************************************************************
 ///	-	DISP&MM-PostProcess
 //$$ MODULE="DISP&MM-PostProcess"
diff --git a/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd b/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd
index d17c9d0eb5a6..9b08ea94ea69 100644
--- a/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd
@@ -156,6 +156,9 @@ void root_func(){
 
 	};
 
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -166,8 +169,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x00a00000>;//10M,you can use 8M if memory isn't enough
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -180,10 +183,21 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-                reserve-memory = <0x02000000>;//32M
-		reserve-iomap = "true";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
 
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
 /// ***************************************************************************************
 ///	-	DISP&MM-PostProcess
 //$$ MODULE="DISP&MM-PostProcess"
diff --git a/arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd b/arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd
index d792e7b43dba..b7a72eb46e66 100644
--- a/arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd
@@ -156,6 +156,9 @@ void root_func(){
         display_size_default = <768 1024 768 3072 32>;//768x1024x4x3==0x0900000 < 0x01000000
 	};
 
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -166,8 +169,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x00a00000>;//10M,you can use 8M if memory isn't enough
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -180,10 +183,21 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-                reserve-memory = <0x04000000>;
-		reserve-iomap = "true";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
 
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
 /// ***************************************************************************************
 ///	-	DISP&MM-PostProcess
 //$$ MODULE="DISP&MM-PostProcess"
diff --git a/arch/arm/boot/dts/amlogic/meson8b_m200_1G.dtd b/arch/arm/boot/dts/amlogic/meson8b_m200_1G.dtd
index ca054c893c59..c1ab925b644b 100644
--- a/arch/arm/boot/dts/amlogic/meson8b_m200_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_m200_1G.dtd
@@ -149,6 +149,9 @@ void root_func(){
 		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -159,7 +162,7 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
+		//reserve-memory = <0x02000000>; // 32M
 		//reserve-iomap = "true";
 	};
 
@@ -173,9 +176,21 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-								reserve-memory = <0x04000000>; // 64M
-//		reserve-iomap = "true";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
 //$$ DEVICE="picdec"
@@ -201,6 +216,7 @@ void root_func(){
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 /// ***************************************************************************************
diff --git a/arch/arm/boot/dts/amlogic/meson8b_m200_1G_iptv.dtd b/arch/arm/boot/dts/amlogic/meson8b_m200_1G_iptv.dtd
index 8eb316d0987c..ad8bd3a6d9da 100644
--- a/arch/arm/boot/dts/amlogic/meson8b_m200_1G_iptv.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_m200_1G_iptv.dtd
@@ -83,7 +83,13 @@ void root_func(){
 		linux,total-memory = <0x3fe00000>;
         #address-cells = <1>;
         #size-cells = <1>;
-
+//$$ DEVICE="mem_4k2k_decoder"
+//$$ L3 PROP_U32 2 ="reg"
+		mem_4k2k_decoder: region@1 {
+			region_name = "cma_4k2k";
+			linux,contiguous-region;
+			reg = <0x00000000 0x0ca00000>;
+		};
         cma_0:region@0 {
            region_name = "cma_0";
            reg = <0 0x01200000>;
@@ -148,6 +154,9 @@ void root_func(){
 		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -158,8 +167,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -172,9 +181,20 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-                reserve-memory = <0x04000000>; // 64M
-                reserve-iomap = "true";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
 //$$ DEVICE="picdec"
diff --git a/arch/arm/boot/dts/amlogic/meson8b_m201_1G.dtd b/arch/arm/boot/dts/amlogic/meson8b_m201_1G.dtd
index e72a0477d999..12a7bdfea26d 100644
--- a/arch/arm/boot/dts/amlogic/meson8b_m201_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_m201_1G.dtd
@@ -84,7 +84,13 @@ void root_func(){
 		linux,total-memory = <0x3fe00000>;
         #address-cells = <1>;
         #size-cells = <1>;
-
+//$$ DEVICE="mem_4k2k_decoder"
+//$$ L3 PROP_U32 2 ="reg"
+		mem_4k2k_decoder: region@1 {
+			region_name = "cma_4k2k";
+			linux,contiguous-region;
+			reg = <0x00000000 0xbc00000>;
+		};
         cma_0:region@0 {
            region_name = "cma_0";
            reg = <0 0x01200000>;
@@ -150,6 +156,9 @@ void root_func(){
 		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -160,7 +169,7 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
+		//reserve-memory = <0x02000000>; // 32M
 		//reserve-iomap = "true";
 	};
 
@@ -174,9 +183,20 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-								reserve-memory = <0x04000000>; // 64M
-//		reserve-iomap = "true";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
 //$$ DEVICE="picdec"
diff --git a/arch/arm/boot/dts/amlogic/meson8b_skt.dtd b/arch/arm/boot/dts/amlogic/meson8b_skt.dtd
index 5eee3ef9c5b1..53cff67db7d5 100644
--- a/arch/arm/boot/dts/amlogic/meson8b_skt.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_skt.dtd
@@ -156,7 +156,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
+		//reserve-memory = <0x02000000>; // 32M
+
 		reserve-iomap = "true";
 	};
 
diff --git a/arch/arm/boot/dts/amlogic/meson8m2_n200C_1G.dtd b/arch/arm/boot/dts/amlogic/meson8m2_n200C_1G.dtd
index 9da66d322f89..981cb118673d 100644
--- a/arch/arm/boot/dts/amlogic/meson8m2_n200C_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8m2_n200C_1G.dtd
@@ -158,6 +158,9 @@ void root_func(){
 		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -168,7 +171,7 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
+		//reserve-memory = <0x02000000>; // 32M
 		//reserve-iomap = "true";
 	};
 
@@ -182,10 +185,21 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-								reserve-memory = <0x04000000>; // 64M
-//		reserve-iomap = "true";
-		linux,contiguous-region = <&mem_4k2k_decoder>;
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
 //$$ DEVICE="picdec"
@@ -211,6 +225,7 @@ void root_func(){
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 /// ***************************************************************************************
diff --git a/arch/arm/boot/dts/amlogic/meson8m2_n200C_2G.dtd b/arch/arm/boot/dts/amlogic/meson8m2_n200C_2G.dtd
index 363ead786553..4bc7c615caad 100644
--- a/arch/arm/boot/dts/amlogic/meson8m2_n200C_2G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8m2_n200C_2G.dtd
@@ -151,6 +151,9 @@ void root_func(){
 		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -161,8 +164,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -175,9 +178,21 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-                reserve-memory = <0x10000000>;
-                reserve-iomap = "true";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
 //$$ DEVICE="picdec"
@@ -203,6 +218,7 @@ void root_func(){
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 /// ***************************************************************************************
diff --git a/arch/arm/boot/dts/amlogic/meson8m2_n200_1G.dtd b/arch/arm/boot/dts/amlogic/meson8m2_n200_1G.dtd
index a94bfdf4e40b..534111e0df5a 100644
--- a/arch/arm/boot/dts/amlogic/meson8m2_n200_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8m2_n200_1G.dtd
@@ -169,7 +169,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		///reserve-memory = <0x02000000>; // 32M
+		//reserve-memory = <0x02000000>; // 32M
+
 		//reserve-iomap = "true";
 	};
 
@@ -223,6 +224,7 @@ void root_func(){
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 /// ***************************************************************************************
diff --git a/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd b/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd
index 84fc6e8cd1ea..f8efdb312c60 100644
--- a/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd
@@ -85,6 +85,13 @@ void root_func(){
 
         #address-cells = <1>;
         #size-cells = <1>;
+//$$ DEVICE="mem_4k2k_decoder"
+//$$ L3 PROP_U32 2 ="reg"
+		mem_4k2k_decoder: region@1 {
+			region_name = "cma_4k2k";
+			linux,contiguous-region;
+			reg = <0x00000000 0x0ca00000>;
+		};
 
         cma_0:region@0 {
            region_name = "cma_0";
@@ -152,6 +159,9 @@ void root_func(){
 		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -162,8 +172,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -176,9 +186,21 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-                reserve-memory = <0x10000000>;
-                reserve-iomap = "true";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
 //$$ DEVICE="picdec"
@@ -204,6 +226,7 @@ void root_func(){
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 /// ***************************************************************************************
diff --git a/arch/arm/boot/dts/amlogic/meson8m2_pxp.dtd b/arch/arm/boot/dts/amlogic/meson8m2_pxp.dtd
index f6578b6901ba..a9c279ecb842 100644
--- a/arch/arm/boot/dts/amlogic/meson8m2_pxp.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8m2_pxp.dtd
@@ -148,6 +148,9 @@ void root_func(){
 //		reserve-memory = <0x02300000>;
 //	};
 
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -158,8 +161,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>;
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -168,13 +171,24 @@ void root_func(){
 //$$ DEVICE="vdec"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 4 ="reg"
-//        vdec{
-//                compatible = "amlogic,vdec";
-//                dev_name = "vdec.0";
-//                status = "okay";
-//                reserve-memory = <0x10000000>;
-//                reserve-iomap = "true";
-//        };
+        vdec{
+                compatible = "amlogic,vdec";
+                dev_name = "vdec.0";
+                status = "okay";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
+        };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		//linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
 
 /// ***************************************************************************************
 ///	-	DISP&MM-PostProcess
diff --git a/arch/arm/boot/dts/amlogic/meson8m2_skt.dtd b/arch/arm/boot/dts/amlogic/meson8m2_skt.dtd
index 84c7e5a54842..1056c63ec1b8 100644
--- a/arch/arm/boot/dts/amlogic/meson8m2_skt.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8m2_skt.dtd
@@ -150,6 +150,9 @@ void root_func(){
 		reserve-memory = <0x01800000>; //8x1920x1088x3/2=24M
 	};
 
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -160,8 +163,8 @@ void root_func(){
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -174,10 +177,21 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-                reserve-memory = <0x10000000>;
-                reserve-iomap = "true";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
 
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		//linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
 /// ***************************************************************************************
 ///	-	DISP&MM-PostProcess
 //$$ MODULE="DISP&MM-PostProcess"
diff --git a/arch/arm/boot/dts/amlogic/mesong9bb_n302.dtd b/arch/arm/boot/dts/amlogic/mesong9bb_n302.dtd
index 608ef026c4a4..0ed83bf35157 100644
--- a/arch/arm/boot/dts/amlogic/mesong9bb_n302.dtd
+++ b/arch/arm/boot/dts/amlogic/mesong9bb_n302.dtd
@@ -771,40 +771,53 @@
 
 
 
-/// ****************************************************************************
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+/// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
 //$$ DEVICE="mesonstream"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 4 ="reg"
-	mesonstream {
+	mesonstream{
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
+		//reserve-memory = <0x02000000>; // 32M
 		//reserve-iomap = "true";
 	};
 
-
-
-/// ****************************************************************************
-///	-	DISP&MM-A/V video dec
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V video dec"
 //$$ DEVICE="vdec"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 4 ="reg"
-	vdec {
-		compatible = "amlogic,vdec";
-		dev_name = "vdec.0";
+        vdec{
+                compatible = "amlogic,vdec";
+                dev_name = "vdec.0";
+                status = "okay";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
+        };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
 		status = "okay";
-		share-memory-name = "continuemem0";
-		share-memory-offset = <0x02100000>;
-		share-memory-size = <0x04000000>; //64M
-//		reserve-iomap = "true";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
 	};
 
 
 
+
+
 /// ****************************************************************************
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
diff --git a/arch/arm/boot/dts/amlogic/mesong9bb_n303.dtd b/arch/arm/boot/dts/amlogic/mesong9bb_n303.dtd
index 3c050320df04..ad80370cc6fd 100644
--- a/arch/arm/boot/dts/amlogic/mesong9bb_n303.dtd
+++ b/arch/arm/boot/dts/amlogic/mesong9bb_n303.dtd
@@ -891,40 +891,52 @@
 
 
 
-/// ****************************************************************************
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+/// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
 //$$ DEVICE="mesonstream"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 4 ="reg"
-	mesonstream {
+	mesonstream{
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
+		//reserve-memory = <0x02000000>; // 32M
 		//reserve-iomap = "true";
 	};
 
-
-
-/// ****************************************************************************
-///	-	DISP&MM-A/V video dec
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V video dec"
 //$$ DEVICE="vdec"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 4 ="reg"
-	vdec {
-		compatible = "amlogic,vdec";
-		dev_name = "vdec.0";
+        vdec{
+                compatible = "amlogic,vdec";
+                dev_name = "vdec.0";
+                status = "okay";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
+        };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
 		status = "okay";
-		share-memory-name = "continuemem0";
-		share-memory-offset = <0x02100000>;
-		share-memory-size = <0x04000000>; //64M
-//		reserve-iomap = "true";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
 	};
 
 
 
+
 /// ****************************************************************************
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
diff --git a/arch/arm/boot/dts/amlogic/mesong9bb_n304.dtd b/arch/arm/boot/dts/amlogic/mesong9bb_n304.dtd
index 2e6df61b7901..d4f33f95fe32 100644
--- a/arch/arm/boot/dts/amlogic/mesong9bb_n304.dtd
+++ b/arch/arm/boot/dts/amlogic/mesong9bb_n304.dtd
@@ -761,6 +761,10 @@
 		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
@@ -771,7 +775,7 @@
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
+		//reserve-memory = <0x02000000>; // 32M
 		//reserve-iomap = "true";
 	};
 
@@ -785,11 +789,22 @@
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-                share-memory-name = "continuemem0";
-                share-memory-offset = <0x02100000>;
-                share-memory-size = <0x04000000>; //64M
-//		reserve-iomap = "true";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
+		status = "okay";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
+		linux,contiguous-region = <&mem_4k2k_decoder>;
+	};
+
+
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
 //$$ DEVICE="picdec"
@@ -817,6 +832,7 @@
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 
diff --git a/arch/arm/boot/dts/amlogic/mesong9bb_t826_pxp.dtd b/arch/arm/boot/dts/amlogic/mesong9bb_t826_pxp.dtd
index 60979cd16a13..e2713afc6545 100644
--- a/arch/arm/boot/dts/amlogic/mesong9bb_t826_pxp.dtd
+++ b/arch/arm/boot/dts/amlogic/mesong9bb_t826_pxp.dtd
@@ -747,40 +747,53 @@
 
 
 
-/// ****************************************************************************
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+/// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
 //$$ DEVICE="mesonstream"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 4 ="reg"
-	mesonstream {
+	mesonstream{
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
-
-
-
-/// ****************************************************************************
-///	-	DISP&MM-A/V video dec
-// MODULE = "DISP&MM-A/V video dec"
-// DEVICE = "vdec"
-// L2 PROP_STR = "status"
-// L3 PROP_U32 4 ="reg"
-	vdec {
-		compatible = "amlogic,vdec";
-		dev_name = "vdec.0";
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+//$$ DEVICE="vdec"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 4 ="reg"
+        vdec{
+                compatible = "amlogic,vdec";
+                dev_name = "vdec.0";
+                status = "okay";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
+        };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
 		status = "okay";
-		reserve-memory = <0x10000000>; // 256M
-		reserve-iomap = "true";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
 		linux,contiguous-region = <&mem_4k2k_decoder>;
 	};
 
 
 
+
+
 /// ****************************************************************************
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
diff --git a/arch/arm/boot/dts/amlogic/mesong9bb_t826_skt.dtd b/arch/arm/boot/dts/amlogic/mesong9bb_t826_skt.dtd
index 7450bf094c04..3721e9090b8e 100644
--- a/arch/arm/boot/dts/amlogic/mesong9bb_t826_skt.dtd
+++ b/arch/arm/boot/dts/amlogic/mesong9bb_t826_skt.dtd
@@ -760,40 +760,53 @@
 
 
 
-/// ****************************************************************************
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+/// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
 //$$ DEVICE="mesonstream"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 4 ="reg"
-	mesonstream {
+	mesonstream{
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
-
-
-
-/// ****************************************************************************
-///	-	DISP&MM-A/V video dec
-// MODULE = "DISP&MM-A/V video dec"
-// DEVICE = "vdec"
-// L2 PROP_STR = "status"
-// L3 PROP_U32 4 ="reg"
-	vdec {
-		compatible = "amlogic,vdec";
-		dev_name = "vdec.0";
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+//$$ DEVICE="vdec"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 4 ="reg"
+        vdec{
+                compatible = "amlogic,vdec";
+                dev_name = "vdec.0";
+                status = "okay";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
+        };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
 		status = "okay";
-		reserve-memory = <0x10000000>; // 256M
-		reserve-iomap = "true";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
 		linux,contiguous-region = <&mem_4k2k_decoder>;
 	};
 
 
 
+
+
 /// ****************************************************************************
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
diff --git a/arch/arm/boot/dts/amlogic/mesong9bb_t828_skt.dtd b/arch/arm/boot/dts/amlogic/mesong9bb_t828_skt.dtd
index 6eda74d00e2f..80a5ef73ab21 100644
--- a/arch/arm/boot/dts/amlogic/mesong9bb_t828_skt.dtd
+++ b/arch/arm/boot/dts/amlogic/mesong9bb_t828_skt.dtd
@@ -853,40 +853,53 @@
 
 
 
-/// ****************************************************************************
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+/// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
 //$$ DEVICE="mesonstream"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 4 ="reg"
-	mesonstream {
+	mesonstream{
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
-
-
-
-/// ****************************************************************************
-///	-	DISP&MM-A/V video dec
-// MODULE = "DISP&MM-A/V video dec"
-// DEVICE = "vdec"
-// L2 PROP_STR = "status"
-// L3 PROP_U32 4 ="reg"
-	vdec {
-		compatible = "amlogic,vdec";
-		dev_name = "vdec.0";
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+//$$ DEVICE="vdec"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 4 ="reg"
+        vdec{
+                compatible = "amlogic,vdec";
+                dev_name = "vdec.0";
+                status = "okay";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
+        };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
 		status = "okay";
-		reserve-memory = <0xc000000>; // 64M
-		reserve-iomap = "true";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
 		linux,contiguous-region = <&mem_4k2k_decoder>;
 	};
 
 
 
+
+
 /// ****************************************************************************
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
diff --git a/arch/arm/boot/dts/amlogic/mesong9tv_n210.dtd b/arch/arm/boot/dts/amlogic/mesong9tv_n210.dtd
index b2ea4ff99be1..ab7d9f280597 100644
--- a/arch/arm/boot/dts/amlogic/mesong9tv_n210.dtd
+++ b/arch/arm/boot/dts/amlogic/mesong9tv_n210.dtd
@@ -768,40 +768,53 @@ NON sdio device(means sd/mmc card), other:reserved */
 
 
 
-/// ****************************************************************************
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+/// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
 //$$ DEVICE="mesonstream"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 4 ="reg"
-	mesonstream {
+	mesonstream{
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-//		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+//$$ DEVICE="vdec"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 4 ="reg"
+        vdec{
+                compatible = "amlogic,vdec";
+                dev_name = "vdec.0";
+                status = "okay";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
+        };
 
 
-
-/// ****************************************************************************
-///	-	DISP&MM-A/V video dec
-// MODULE = "DISP&MM-A/V video dec"
-// DEVICE = "vdec"
-// L2 PROP_STR = "status"
-// L3 PROP_U32 4 ="reg"
-	vdec {
-		compatible = "amlogic,vdec";
-		dev_name = "vdec.0";
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
 		status = "okay";
-		reserve-memory = <0x04000000>; // 64M
-//		reserve-iomap = "true";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
 		linux,contiguous-region = <&mem_4k2k_decoder>;
 	};
 
 
 
+
+
 /// ****************************************************************************
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
@@ -830,6 +843,7 @@ NON sdio device(means sd/mmc card), other:reserved */
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
+
 	};
 
 
diff --git a/arch/arm/boot/dts/amlogic/mesong9tv_n300.dtd b/arch/arm/boot/dts/amlogic/mesong9tv_n300.dtd
index 6f3ed1aaa269..2b32529ef01e 100644
--- a/arch/arm/boot/dts/amlogic/mesong9tv_n300.dtd
+++ b/arch/arm/boot/dts/amlogic/mesong9tv_n300.dtd
@@ -775,40 +775,52 @@
 
 
 
-/// ****************************************************************************
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+/// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
 //$$ DEVICE="mesonstream"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 4 ="reg"
-	mesonstream {
+	mesonstream{
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
+		//reserve-memory = <0x02000000>; // 32M
 		//reserve-iomap = "true";
 	};
 
-
-
-
-/// ****************************************************************************
-///	-	DISP&MM-A/V video dec
-// MODULE = "DISP&MM-A/V video dec"
-// DEVICE = "vdec"
-// L2 PROP_STR = "status"
-// L3 PROP_U32 4 ="reg"
-	vdec {
-		compatible = "amlogic,vdec";
-		dev_name = "vdec.0";
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+//$$ DEVICE="vdec"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 4 ="reg"
+        vdec{
+                compatible = "amlogic,vdec";
+                dev_name = "vdec.0";
+                status = "okay";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
+        };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
 		status = "okay";
-		reserve-memory = <0x04000000>; // 64M
+		reserve-memory = <0x04100000>; // 65M
 		//reserve-iomap = "true";
 		linux,contiguous-region = <&mem_4k2k_decoder>;
 	};
 
 
 
+
 /// ****************************************************************************
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
diff --git a/arch/arm/boot/dts/amlogic/mesong9tv_n300_15g.dtd b/arch/arm/boot/dts/amlogic/mesong9tv_n300_15g.dtd
index e0f4dcd115a6..525a47ce6fe7 100644
--- a/arch/arm/boot/dts/amlogic/mesong9tv_n300_15g.dtd
+++ b/arch/arm/boot/dts/amlogic/mesong9tv_n300_15g.dtd
@@ -763,40 +763,53 @@
 
 
 
-/// ****************************************************************************
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+/// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
 //$$ DEVICE="mesonstream"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 4 ="reg"
-	mesonstream {
+	mesonstream{
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
+		//reserve-memory = <0x02000000>; // 32M
 		//reserve-iomap = "true";
 	};
 
-
-
-
-/// ****************************************************************************
-///	-	DISP&MM-A/V video dec
-// MODULE = "DISP&MM-A/V video dec"
-// DEVICE = "vdec"
-// L2 PROP_STR = "status"
-// L3 PROP_U32 4 ="reg"
-	vdec {
-		compatible = "amlogic,vdec";
-		dev_name = "vdec.0";
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+//$$ DEVICE="vdec"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 4 ="reg"
+        vdec{
+                compatible = "amlogic,vdec";
+                dev_name = "vdec.0";
+                status = "okay";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
+        };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
 		status = "okay";
-		reserve-memory = <0x04000000>; // 64M
+		reserve-memory = <0x04100000>; // 65M
 		//reserve-iomap = "true";
 		linux,contiguous-region = <&mem_4k2k_decoder>;
 	};
 
 
 
+
+
 /// ****************************************************************************
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
diff --git a/arch/arm/boot/dts/amlogic/mesong9tv_n300_1g.dtd b/arch/arm/boot/dts/amlogic/mesong9tv_n300_1g.dtd
index f6ce0a6f69cd..121a715f77e7 100644
--- a/arch/arm/boot/dts/amlogic/mesong9tv_n300_1g.dtd
+++ b/arch/arm/boot/dts/amlogic/mesong9tv_n300_1g.dtd
@@ -763,40 +763,53 @@
 
 
 
-/// ****************************************************************************
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+/// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
 //$$ DEVICE="mesonstream"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 4 ="reg"
-	mesonstream {
+	mesonstream{
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-//		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
-
-
-
-/// ****************************************************************************
-///	-	DISP&MM-A/V video dec
-// MODULE = "DISP&MM-A/V video dec"
-// DEVICE = "vdec"
-// L2 PROP_STR = "status"
-// L3 PROP_U32 4 ="reg"
-	vdec {
-		compatible = "amlogic,vdec";
-		dev_name = "vdec.0";
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+//$$ DEVICE="vdec"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 4 ="reg"
+        vdec{
+                compatible = "amlogic,vdec";
+                dev_name = "vdec.0";
+                status = "okay";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
+        };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
 		status = "okay";
-		reserve-memory = <0x04000000>; // 64M
-//		reserve-iomap = "true";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
 		linux,contiguous-region = <&mem_4k2k_decoder>;
 	};
 
 
 
+
+
 /// ****************************************************************************
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
diff --git a/arch/arm/boot/dts/amlogic/mesong9tv_n301.dtd b/arch/arm/boot/dts/amlogic/mesong9tv_n301.dtd
index 58522e0e8526..ead64d2f67ce 100644
--- a/arch/arm/boot/dts/amlogic/mesong9tv_n301.dtd
+++ b/arch/arm/boot/dts/amlogic/mesong9tv_n301.dtd
@@ -208,40 +208,52 @@
 
 
 
-/// ****************************************************************************
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+/// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
 //$$ DEVICE="mesonstream"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 4 ="reg"
-	mesonstream {
+	mesonstream{
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
+		//reserve-memory = <0x02000000>; // 32M
 		//reserve-iomap = "true";
 	};
 
-
-
-
-/// ****************************************************************************
-///	-	DISP&MM-A/V video dec
-// MODULE = "DISP&MM-A/V video dec"
-// DEVICE = "vdec"
-// L2 PROP_STR = "status"
-// L3 PROP_U32 4 ="reg"
-	vdec {
-		compatible = "amlogic,vdec";
-		dev_name = "vdec.0";
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+//$$ DEVICE="vdec"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 4 ="reg"
+        vdec{
+                compatible = "amlogic,vdec";
+                dev_name = "vdec.0";
+                status = "okay";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
+        };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
 		status = "okay";
-		reserve-memory = <0x04000000>; // 64M
+		reserve-memory = <0x04100000>; // 65M
 		//reserve-iomap = "true";
 		linux,contiguous-region = <&mem_4k2k_decoder>;
 	};
 
 
 
+
 /// ****************************************************************************
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
diff --git a/arch/arm/boot/dts/amlogic/mesong9tv_pxp.dtd b/arch/arm/boot/dts/amlogic/mesong9tv_pxp.dtd
index b19941c28259..b739863fd6aa 100644
--- a/arch/arm/boot/dts/amlogic/mesong9tv_pxp.dtd
+++ b/arch/arm/boot/dts/amlogic/mesong9tv_pxp.dtd
@@ -473,40 +473,52 @@
 
 
 
-/// ****************************************************************************
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+/// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
 //$$ DEVICE="mesonstream"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 4 ="reg"
-	mesonstream {
+	mesonstream{
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
-
-
-
-/// ****************************************************************************
-///	-	DISP&MM-A/V video dec
-// MODULE = "DISP&MM-A/V video dec"
-// DEVICE = "vdec"
-// L2 PROP_STR = "status"
-// L3 PROP_U32 4 ="reg"
-	vdec {
-		compatible = "amlogic,vdec";
-		dev_name = "vdec.0";
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+//$$ DEVICE="vdec"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 4 ="reg"
+        vdec{
+                compatible = "amlogic,vdec";
+                dev_name = "vdec.0";
+                status = "okay";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
+        };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
 		status = "okay";
-		reserve-memory = <0x04000000>; // 64M
-		reserve-iomap = "true";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
 		linux,contiguous-region = <&mem_4k2k_decoder>;
 	};
 
 
 
+
 /// ****************************************************************************
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
diff --git a/arch/arm/boot/dts/amlogic/mesong9tv_skt.dtd b/arch/arm/boot/dts/amlogic/mesong9tv_skt.dtd
index f174872f103d..90185f07b8e6 100644
--- a/arch/arm/boot/dts/amlogic/mesong9tv_skt.dtd
+++ b/arch/arm/boot/dts/amlogic/mesong9tv_skt.dtd
@@ -627,40 +627,53 @@
 
 
 
-/// ****************************************************************************
+
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+/// ***************************************************************************************
 ///	-	DISP&MM-A/V stream
 //$$ MODULE = "DISP&MM-A/V stream"
 //$$ DEVICE="mesonstream"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 4 ="reg"
-	mesonstream {
+	mesonstream{
 		compatible = "amlogic,mesonstream";
 		dev_name = "mesonstream.0";
 		status = "okay";
-		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-memory = <0x02000000>; // 32M
+		//reserve-iomap = "true";
 	};
 
-
-
-
-/// ****************************************************************************
-///	-	DISP&MM-A/V video dec
-// MODULE = "DISP&MM-A/V video dec"
-// DEVICE = "vdec"
-// L2 PROP_STR = "status"
-// L3 PROP_U32 4 ="reg"
-	vdec {
-		compatible = "amlogic,vdec";
-		dev_name = "vdec.0";
+/// ***************************************************************************************
+///     -       DISP&MM-A/V stream
+//$$ MODULE = "DISP&MM-A/V video dec"
+//$$ DEVICE="vdec"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 4 ="reg"
+        vdec{
+                compatible = "amlogic,vdec";
+                dev_name = "vdec.0";
+                status = "okay";
+///				reserve-memory = <0x04000000>; // 64M
+//				reserve-iomap = "true";
+//				linux,contiguous-region = <&mem_4k2k_decoder>;
+        };
+
+
+	codec_mm{
+		compatible = "amlogic, codec, mm";
+		dev_name = "codec_mm";
 		status = "okay";
-		reserve-memory = <0x10000000>; // 256M
-		reserve-iomap = "true";
+		reserve-memory = <0x04100000>; // 65M
+		//reserve-iomap = "true";
 		linux,contiguous-region = <&mem_4k2k_decoder>;
 	};
 
 
 
+
+
 /// ****************************************************************************
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
-- 
2.19.0

